tests/integration/basket/test_views.py

```diff
@@ -109,3 +111,36 @@ def test_default_shipping_address_for_anonymous_user(self):
         request = RequestFactory().get(self.url)
         view = views.BasketView(request=request)
         self.assertIsNone(view.get_default_shipping_address())
+
+
+class TestVoucherViews(CheckoutMixin, WebTestCase):
+    csrf_checks = False
+
+    def setUp(self):
+        self.voucher = factories.create_voucher()
+        super().setUp()
+
+    def test_add_voucher(self):
+        """
+        Checks that voucher can be added to basket through appropriate view.
+        """
+        self.add_product_to_basket()
+
+        assert self.voucher.basket_set.count() == 0
+
+        response = self.post(reverse('basket:vouchers-add'), params={'code': self.voucher.code})
+        assert response.status_code == 302  # Redirect to basket summary page
```

Comment:
We can use some dedicated methods for this:

```python
self.assertIsRedirect(response)
self.assertRedirectsTo(response, 'basket:index')
```

Reply:
Yes, you are right. Thanks!